UnicornRender/video/include/unicorn/video/Primitives.hpp

```diff
@@ -10,41 +10,54 @@
 #include <unicorn/utility/SharedMacros.hpp>
 #include <unicorn/video/Mesh.hpp>
 
+#include <string>
+#include <list>
+
 namespace unicorn
 {
 namespace video
 {
-/** @brief  Fills meshes with geometric primitive's data */
+/** @brief  Fills meshes with geometric primitives data */
 class Primitives
 {
 public:
     /**
     *  @brief Fills given mesh with cube data
     *
-    *  @param[out] mesh    geometry mesh
-    *  @return reference to updated mesh
+    *  @param[in, out] mesh geometry mesh
+    *  @return pointer to filled mesh
     */
-    UNICORN_EXPORT static Mesh& Box(Mesh& mesh);
+    UNICORN_EXPORT static Mesh* Box(Mesh* mesh = nullptr);
```

Comment:
why pointer? This method just fills the values, why do you want also to let it create new objects? I would suggest leaving it as it was and letting caller decide where and how the mesh is created